samples/csharp/end-to-end-apps/ObjectDetection-Onnx/README.md

```diff
@@ -0,0 +1,194 @@
+
+```csharp
+public class ImageInputData
+{
+    [ImageType(416, 416)]
+    public Bitmap Image { get; set; }
+}
+```
+
+### ML.NET: Configure the model
+
+The first step is to create an empty DataView as we just need schema of data while configuring up model.
+
+```csharp
+var dataView = _mlContext.Data.LoadFromEnumerable(new List<ImageInputData>());
+```
+
+The second step is to define the estimator pipeline. Usually when dealing with deep neural networks, you must adapt the images to the format expected by the network. For this reason, the code below resizes and transforms the images (pixel values are normalized across all R,G,B channels).
+
+```csharp
+var pipeline = _mlContext.Transforms.ResizeImages(resizing: ImageResizingEstimator.ResizingKind.Fill, outputColumnName: "image", imageWidth: ImageSettings.imageWidth, imageHeight: ImageSettings.imageHeight, inputColumnName: nameof(ImageInputData.Image))
+                .Append(_mlContext.Transforms.ExtractPixels(outputColumnName: "image"))
+                .Append(_mlContext.Transforms.ApplyOnnxModel(modelFile: onnxModelFilePath, outputColumnNames: new[] { TinyYoloModelSettings.ModelOutput }, inputColumnNames: new[] { TinyYoloModelSettings.ModelInput }));
+```
+
+You also need to inspect the neural network to get the names of the input and output nodes. To do this, you can use tools like [Netron](https://github.com/lutzroeder/netron), which is automatically installed with [Visual Studio Tools for AI](https://visualstudio.microsoft.com/downloads/ai-tools-vs/).
+These names are used later when we define the estimation pipeline: in the case of the Tiny Yolo2 network, the input tensor is named **'image'** and the output is named **'grid'**.
+
+We'll use these to define the **input** and **output** parameters of the Tiny Yolo2 Onnx Model.
+
+```csharp
+public struct TinyYoloModelSettings
+{
+    // for checking TIny yolo2 Model input and output parameter names,
+    // you can use tools like Netron, which is installed by Visual Studio AI Tools
+
+    // input tensor name
+    public const string ModelInput = "image";
+
+    // output tensor name
+    public const string ModelOutput = "grid";
+}
+```
+
+![inspecting neural network with netron](./docs/Netron/netron.PNG)
+
+Create the model by fitting the DataView.
+
+```csharp
+var model = pipeline.Fit(dataView);
+```
+
+## Load model and create PredictionEngine
+
+After the model is configured, we need to save the model, load the saved model, create a `PredictionEngine`, and the pass the image to the engine to detect objects using the model.
+This is one place that the **Web** app and the **WPF** app differ slightly.  The **Web** app uses a `PredicitonEnginePool` to efficiently manage and provide the service with a `PredictionEngine` to use to make predictions.
```

Comment:
Separate out the two sentences so that users can visually see a separation between the web app vs desktop app explanations.

```suggestion
This is one place that the **Web** app and the **WPF** app differ slightly.  

The **Web** app uses a `PredicitonEnginePool` to efficiently manage and provide the service with a `PredictionEngine` to use to make predictions.
```